pkg/operator/bootstrap.go

```diff
@@ -235,38 +235,20 @@ func appendManifestsByPlatform(manifests []manifest, infra configv1.Infrastructu
 	}
 
 	if infra.Status.PlatformStatus.VSphere != nil {
-		deployInternalLB := true
-		// vSphere allows to use a user managed load balancer by not setting the VIPs in PlatformStatus.
-		// We will maintain backward compatibility by checking if the VIPs are not set, we will not deploy
-		// Keepalived and CoreDNS.
+		// TODO(mko) It is not clear why for user-managed LB and for ELB we want to skip CoreDNS as every
```

Comment:
I believe there was some related logic implemented for VSphere that overlapped the external loadbalancer feature. I don't remember all of the details off hand, but that's why there is VSphere-specific code here.